src/popover/use-popover-position.ts

```diff
@@ -63,10 +63,14 @@ export default function usePopoverPosition({
       const document = popover.ownerDocument;
       const track = trackRef.current;
 
+      const rootNode = popover.getRootNode();
+      const isInShadowDom = rootNode !== document && rootNode.nodeType === Node.DOCUMENT_FRAGMENT_NODE;
```

Comment:
I think this check `rootNode !== document`  might result in false positives when popover is rendered within an iframe, in which case the getRootNode() will return the iframe's document, not top document.

Reply:
I will have to double-check this, but my first thought is that the iframe's document wouldn't be a `Node.DOCUMENT_FRAGMENT_NODE`.

Reply:
You're right. I haven't realised that is not the global document but `popover.ownerDocument`. I think it would be nice to add a new util to utils/dom, something like:

```
function isInShadowDom(element: Element) {
  const rootNode = element.getRootNode();
  return rootNode !== element.ownerDocument && rootNode.nodeType === Node.DOCUMENT_FRAGMENT_NODE;
}
```